test/decorators/provider_decorator_test.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ProviderDecoratorTest < Draper::TestCase
+
+  def setup
+    provider = FactoryBot.create(:provider_account)
+    @decorator = ProviderDecorator.new provider
+    @decorator.stubs(:service_plans_management_visible?).returns(true)
+  end
+
+  test 'new_application_form_data in Dashboard context' do
+    data = {}
+    @decorator.expects(:audience_context_new_application_form_data).returns(data).once
+
+    assert @decorator.new_application_form_data
+  end
+
+  test 'new_application_form_data in Account context' do
+    buyer = FactoryBot.create(:buyer_account)
+    data = {}
+    @decorator.expects(:buyer_context_new_application_form_data).with(buyer).returns(data).once
+
+    assert @decorator.new_application_form_data(buyer: buyer)
+  end
+
+  test 'new_application_form_data in Product context' do
+    service = FactoryBot.create(:service)
+    data = {}
+    @decorator.expects(:product_context_new_application_form_data).with(service).returns(data).once
+
+    assert @decorator.new_application_form_data(service: service)
+  end
+
+  test 'new_application_form_data inline errors' do
+    cinstance = FactoryBot.create(:cinstance)
+    errors = ['Name is required']
+    cinstance.stubs(:errors).returns(errors)
+    @decorator.stubs(:audience_context_new_application_form_data).returns({}).once
```

Comment:
stubs once does not really do anything. Stub is always and should not expect any other modifier and does not expect/raise anything. `expects` raises if the expectations are not met

https://github.com/freerange/mocha#quick-start

Reply:
Oh I missed this one, it's fixed now